subsys/net/lib/http/http_client.c

```diff
@@ -438,24 +464,28 @@ static int http_wait_data(int sock, struct http_request *req, int32_t timeout)
 		ret = zsock_poll(fds, nfds, remaining_time);
 		if (ret == 0) {
 			LOG_DBG("Timeout");
-			goto finalize_data;
+			ret = -ETIMEDOUT;
+			goto error;
 		} else if (ret < 0) {
+			ret = -errno;
 			goto error;
 		}
 		if (fds[0].revents & (ZSOCK_POLLERR | ZSOCK_POLLNVAL)) {
+			ret = -errno;
 			goto error;
 		} else if (fds[0].revents & ZSOCK_POLLHUP) {
 			/* Connection closed */
 			LOG_DBG("Connection closed");
```

Comment:
can this and the one below be moved to the `closed:` label?

Reply:
Hypothetically, yes, and I started to do that, but I ultimately thought the code was a little easier to read this way.

I don't care a ton either way, but if you also don't care too much I will leave as-is to speed up review/merge